website/docs/r/ssm_association.html.markdown

```diff
@@ -51,6 +51,20 @@ Targets specify what instance IDs or tags to apply the document to and has these
 * `key` - (Required) Either `InstanceIds` or `tag:Tag Name` to specify an EC2 tag.
 * `values` - (Required) A list of instance IDs or tag values. AWS currently limits this list size to one value.
 
+-> **Note:** To target all instances in an AWS account, set the `key` as `"InstanceIds"` with `values` set as `["*"]`.
+Example:
+
+```terraform
+resource "aws_ssm_association" "example" {
+  name = "AmazonCloudWatch-ManageAgent"
+
+  targets {
+    key    = "InstanceIds"
+    values = ["*"]
+  }
+}
+```
+
```

Comment:
I think it probably makes sense to move this up under the Examples header and switch from using `**Note**` to an H3 header; perhaps something like `### Targeting All Instances`. Note that that would also require adding an H3 header to the existing example (perhaps `### Targeting A Single Instance`?). 

There's precedence for this in a few places. I grabbed the [aws_s3_bucket](https://github.com/hashicorp/terraform-provider-aws/blob/de0147b7c2d9051bee842b8f6579e9dbbbf831bf/website/docs/r/s3_bucket.html.markdown#L15-L55) documentation as an example.

Reply:
Thanks @justinretzolk for the review! I've made the requested changes. For completeness I've also added an example on how to target **managed** instances based on their tags.